arrows/vtk/mesh_coloration.cxx

```diff
@@ -649,7 +650,7 @@ ::render_depth_buffer(
   // we only render points in front of the camera
   if( depthRange[ 0 ] < 0 )
   {
-    depthRange[ 0 ] = 0;
+    depthRange[ 0 ] = 0.01;
```

Comment:
This is potentially problematic for a several reasons.  First, this value of `0.01` only kicks in for values less than zero.  So if the actual near depth is zero this offset is not applied.  The same is true for any values less than 0.01.  To correct this hack we also need to change line 650 to `if( depthRange[ 0 ] < 0.01 )`.

However, even with the above change, this 0.01 is very arbitrary and makes assumptions about the scale of the scene.  This will not work for a scene where the overall scale is very small.  The "right" way to do this is to project all the vertices and check all vertices in front of the camera that lie in the image.  But that is expensive.  An alternate hack that is likely safer than 0.01 is to scale that threshold relative to `depthRange[ 1 ]`, so maybe ` depthRange[ 0 ] = depthRange[ 1 ] * 0.01` or similar.

Reply:
Sounds good.